backend/src/api/comment/create_comment.rs

```diff
@@ -34,11 +40,51 @@ pub(super) async fn post<D: SqliteDatabase>(
     .await
 }
 
+async fn extract_alias<D: SqliteDatabase>(
+    mut text: String,
+    state: &Arc<AppState<D>>,
+) -> anyhow::Result<String> {
+    let aliases: Vec<String> = text
+        .split(" ")
+        .filter(|word| word.starts_with("!"))
+        .map(|word| word[1..].to_string())
+        .filter(|word| !word.is_empty())
+        .collect();
+
+    for alias in aliases {
+        // For each alias, fetch the link
+        //
+        // If link is available, replace alias with link
+        //
+        // If no alias is available, skip
+        let cloned = alias.clone();
+        let conn = state.pool.get().await.context("Failed to get connection")?;
```

Comment:
You can move the connection outside the `for` loop so we dont create a new DB connection for every alias we want to replace

Reply:
Since the function only uses the connection from the state passing in just the connection instead of the state would give the calling code more flexibility by allowing it to reusing a connection.